### PR TITLE
feat: use kvstore to fetch gas per blob byte

### LINKED ISSUE #1001: Replace GasPerBlobByte const with stored param
### Summary

As title

### PR DESCRIPTION
**Pull request by rahulghangas**
Replaces gas per blob byte constant with stored param

- [x] closes #1001

-- Note - To be merged after #1031

### CODE REVIEW

File: x/blob/keeper/keeper.go [hunk 1]
```diff
@@ -58,7 +53,9 @@ func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 func (k Keeper) PayForBlob(goCtx context.Context, msg *types.MsgPayForBlob) (*types.MsgPayForBlobResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
-	gasToConsume := uint64(shares.MsgSharesUsed(int(msg.BlobSize)) * GasPerMsgShare)
+	// calculate gas per message share by fetching the constant share size and the gas cost per byte from the KV store
+	gasPerMsgShare := appconsts.ShareSize * k.GasPerBlobByte(ctx)
+	gasToConsume := uint64(shares.MsgSharesUsed(int(msg.BlobSize)) * int(gasPerMsgShare))
```
Comment (rootulp) [inline x/blob/keeper/keeper.go:58]: [question] Is it possible to include a unit test that verifies the gas consumed now depends on the param? Perhaps the unit test can determine the gas cost of transaction `k.GasPerBlobByte(ctx) = 8` and `k.GasPerBlobByte(ctx) = 9` and verify the latter is > the former.

If a unit test is is difficult to construct, can this please be manually test to confirm it works as expected?

**Comment by evan-forbes** [comment]
nice! what's left for this? is this blocked by anything?

File: x/blob/types/params.go [hunk 2]
```diff
@@ -3,25 +3,26 @@ package types
...
 	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
 	"gopkg.in/yaml.v2"
 )
 
 var _ paramtypes.ParamSet = (*Params)(nil)
 
 var (
-	KeyMinSquareSize            = []byte("MinSquareSize")
-	DefaultMinSquareSize uint32 = 1
+	KeyMinSquareSize            = []byte("DefaultMinSquareSize")
```
Comment (cmwaters) [inline x/blob/types/params.go:14]: Does this variable really need to add the `Default` prefix? Same for `KeyMaxSquareSize`

Comment (rahulghangas) [reply x/blob/types/params.go:14]: whoops, leftover bugs from the appconsts PR, I'll refactor

**Review by evan-forbes — APPROVED** [review]
great work! thanks for adding the really good tests!!

File: x/blob/keeper/gas_test.go [hunk 3]
```diff
@@ -21,28 +58,51 @@ func TestPayForBlobGas(t *testing.T) {
...
 			name:            "100 byte blob", // occupies 1 share
 			msg:             types.MsgPayForBlob{BlobSize: 100},
-			wantGasConsumed: uint64(4096),
+			wantGasConsumed: uint64(5156),
 		},
 		{
 			name:            "1024 byte blob", // occupies 3 shares because share prefix (e.g. namespace, info byte)
 			msg:             types.MsgPayForBlob{BlobSize: 1024},
-			wantGasConsumed: uint64(12288), // 3 shares * 512 bytes per share * 8 gas per byte = 12288 gas
+			wantGasConsumed: uint64(13348), // 3 shares * 512 bytes per share * 8 gas per byte + 1060 gas for fetching param = 12288 gas
```
Comment (rootulp) [inline x/blob/keeper/gas_test.go:71]: the sum wasn't updated in the comment
```go
			wantGasConsumed: uint64(13348), // 3 shares * 512 bytes per share * 8 gas per byte + 1060 gas for fetching param = 13348 gas
```

File: x/blob/keeper/keeper.go [hunk 4]
```diff
@@ -58,7 +52,9 @@ func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 func (k Keeper) PayForBlob(goCtx context.Context, msg *types.MsgPayForBlob) (*types.MsgPayForBlobResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
-	gasToConsume := uint64(shares.BlobSharesUsed(int(msg.BlobSize)) * GasPerBlobShare)
+	// calculate gas per message share by fetching the constant share size and the gas cost per byte from the KV store
+	gasPerMsgShare := appconsts.ShareSize * k.GasPerBlobByte(ctx)
```
Comment (rootulp) [inline x/blob/keeper/keeper.go:56]: [proposal]
message => blob

**Pull request merged**